### ISSUE TITLE
more glmer.nb evaluation stuff

### ISSUE THREAD

**Issue by bbolker** [lme4/lme4#322]
There are some more problems with evaluation/passage of arguments in `glmer.nb` when other optional arguments are given.

```
set.seed(101)
dd <- expand.grid(f1 = factor(1:3),
                  f2 = LETTERS[1:2], g=1:9, rep=1:15,
                  KEEP.OUT.ATTRS=FALSE)
mu <- 5*(-4 + with(dd, as.integer(f1) + 4*as.numeric(f2)))
dd$y <- rnbinom(nrow(dd),mu=mu,size=1)
library(ggplot2)
ggplot(dd,aes(f1,y))+stat_sum()
m.nb2 <- glmer.nb(y~f1+(1|g),
                                    data=dd,
                                    ## offset=rep(0,nrow(dd)),
                                    contrasts=list(f1=contr.sum))
m.nb2 <- glmer.nb(y~f1+(1|g),
                                    data=dd,
                                    ## offset=rep(log(2),nrow(dd)),
                                    contrasts=list(f1=contr.sum))
## Error in eval(expr, envir, enclos) : 
##  ..3 used in an incorrect context, no ... to look in
```
**update**: works better, but still fails (with `Downdated VtV is not positive definite`) when the offset is included, *even when it's `offset(rep(0,nrow(dd)))` ...

And another related point: if `data` is not specified by name, it doesn't end up in the `@call` slot:
```
m.nb2 <- glmer.nb(y~f1+(1|g), dd)
m.nb2@call
glmer(formula = y ~ f1 + (1 | g), family = negative.binomial(theta = 0.42506227914918))
```

**Comment by bbolker** [comment]
We can probably address  #285 at the same time ...

**Comment by bbolker** [comment]
This stuff is partly resolved by fcde0d1 , but offsets still don't work (even trivial ones):

```
       m.nb2 <- glmer.nb(y~f1+(1|g),
                         data=dd,
                         offset=rep(0,nrow(dd)))
```
fails with a "Downdated VtV error" ...

**Comment by bbolker** [comment]
hopefully fixed by 3c0fa76 !
